sklearn/utils/fixes.py

```diff
@@ -193,3 +193,54 @@ def __getstate__(self):
                                  self._fill_value)
 else:
     from numpy.ma import MaskedArray    # noqa
+
+
+if np_version < (1, 9, 0):
+    # Allow unique to return counts
+    # https://github.com/numpy/numpy/commit/da3c6a28f651877dfb4cfdbbbe76dab985669252
+
+    def unique(ar, return_index=False, return_inverse=False,
+               return_counts=False):
+        ar = np.asanyarray(ar).flatten()
+
+        optional_indices = return_index or return_inverse
+        optional_returns = optional_indices or return_counts
+
+        if ar.size == 0:
+            if not optional_returns:
+                ret = ar
+            else:
+                ret = (ar,)
+                if return_index:
+                    ret += (np.empty(0, np.bool),)
+                if return_inverse:
+                    ret += (np.empty(0, np.bool),)
+                if return_counts:
+                    ret += (np.empty(0, np.intp),)
+            return ret
+
+        if optional_indices:
+            perm = ar.argsort(
+                kind='mergesort' if return_index else 'quicksort')
+            aux = ar[perm]
+        else:
+            ar.sort()
+            aux = ar
+        flag = np.concatenate(([True], aux[1:] != aux[:-1]))
+
+        if not optional_returns:
+            ret = aux[flag]
+        else:
+            ret = (aux[flag],)
+            if return_index:
+                ret += (perm[flag],)
+            if return_inverse:
+                iflag = np.cumsum(flag) - 1
+                iperm = perm.argsort()
```

Comment:
This part is not up-to-date with numpy since https://github.com/numpy/numpy/commit/036efee0caf0bf026a9e8a9fd2010be11c7d9408

Reply:
Okay, but this change was in numpy 1.10.0b1. Shouldn't the 1.9.X version be backported? Otherwise why not backport the current version 1.13.3 that also has an `axis` keyword argument?

Reply:
Indeed, the function from `1.9.X` might be more likely to work with versions < `1.9`.
Yet, if the function from `1.13.3` works in the tests, it should be better, since a bug fix and an speed improvement have been added.

What do you think?

Reply:
I think the axis feature is pretty useful so I would be for including it if it passes the tests.

Reply:
I don't know though. Is it common to backport something from numpy's master branch instead of a maintenance branch?